InjectionIII/AppDelegate.swift

```diff
@@ -0,0 +1,218 @@
+//
+//  AppDelegate.swift
+//  InjectionIII
+//
+//  Created by John Holdsworth on 06/11/2017.
+//  Copyright © 2017 John Holdsworth. All rights reserved.
+//
+
+import Cocoa
+
+var appDelegate: AppDelegate!
+
+@NSApplicationMain
+class AppDelegate : NSObject, NSApplicationDelegate {
+
+    @IBOutlet var window: NSWindow!
+    @IBOutlet weak var enableWatcher: NSMenuItem!
+    @IBOutlet weak var traceItem: NSMenuItem!
+    @IBOutlet weak var statusMenu: NSMenu!
+    @IBOutlet weak var startItem: NSMenuItem!
+    @IBOutlet weak var xprobeItem: NSMenuItem!
+    @IBOutlet weak var enabledTDDItem: NSMenuItem!
+    @IBOutlet weak var enableVaccineItem: NSMenuItem!
+    @IBOutlet weak var windowItem: NSMenuItem!
+    @IBOutlet weak var frontItem: NSMenuItem!
+    @IBOutlet var statusItem: NSStatusItem!
+
+    var watchedDirectories = Set<String>()
+    weak var lastConnection: InjectionServer?
+    var selectedProject: String?
+
+    @objc func applicationDidFinishLaunching(_ aNotification: Notification) {
+        // Insert code here to initialize your application
+        appDelegate = self
+        InjectionServer.startServer(INJECTION_ADDRESS)
+
+        let statusBar = NSStatusBar.system
+        statusItem = statusBar.statusItem(withLength: statusBar.thickness)
+        statusItem.toolTip = "Code Injection"
+        statusItem.highlightMode = true
+        statusItem.menu = statusMenu
+        statusItem.isEnabled = true
+        statusItem.title = ""
+
+        enabledTDDItem.state = UserDefaults.standard.bool(forKey:UserDefaultsTDDEnabled)
+            ? NSControl.StateValue.on
+            : NSControl.StateValue.off
+        enableVaccineItem.state = UserDefaults.standard.bool(forKey:UserDefaultsVaccineEnabled)
+            ? NSControl.StateValue.on
+            : NSControl.StateValue.off
+
+        setMenuIcon("InjectionIdle")
+        DDHotKeyCenter.shared()?
+            .registerHotKey(withKeyCode: UInt16(kVK_ANSI_Equal),
+               modifierFlags: NSEvent.ModifierFlags.control.rawValue,
+               target:self, action:#selector(autoInject(_:)), object:nil)
+    }
+
+    @IBAction func openProject(_ sender: Any) {
+        _ = application(NSApp, openFile:"")
+    }
+
+    @IBAction func addProject(_ sender: Any) {
+        let open = NSOpenPanel()
+        open.prompt = NSLocalizedString("Add Project Directory", tableName: "Project Directory", comment: "Project Directory")
+        open.canChooseDirectories = true
+        open.canChooseFiles = true
+        if open.runModal().rawValue == NSFileHandlingPanelOKButton {
```

Comment:
Do we need to use the `rawValue` here or could we do something like this?

```swift
if open.runModal() == NSApplication.ModalResponse.OK {
```

A bit unsure about the type inference here but we might be able to do this:

```swift
if open.runModal() == .OK {
```

Reply:
Ahaaa....